lib/src/over_react_test/prop_type_utils.dart

```diff
@@ -0,0 +1,87 @@
+import 'dart:html';
+import 'dart:js';
+
+import 'package:matcher/matcher.dart';
+import 'package:meta/meta.dart';
+import 'package:over_react/component_base.dart';
+import 'package:over_react_test/jacket.dart';
+import 'package:react/react_client/react_interop.dart' show PropTypes;
+import 'package:test/test.dart';
+
+/// Tests the prop types of a component by calling the props, adding children if necessary, and checking the console
+/// for expected log output.
+void testPropTypesWithUiProps(
+    {@required UiProps componentProps, dynamic childProps, String customErrorMessage, Element mountNode}) {
```

Comment:
Also more than likely people will have multiple propTypes per component so we should probably also make the `customErrorMessage` into `errorMessages` or `errorMatchers` and either be a map that would be structured like `{propName: error/errorMatcher}` or a list of error strings that are possible (i prefer the map)

Reply:
shouldn't `UiProps  componentProps` technically be `UiFactory factory` cause you'd pass in like `Button` right?

Reply:
On the first point - I see what mean and made `customErrorMessage` into a `List`. I went with a `List` because a `Map` would have required more logic to determine which `key` is being tested and then accessing the `value`. A simple way around that is just to make the order of the `keys` being important, but then a `List` seemed simpler. That's ultimately the route I went with! I also stuck with `String`s and not `propErrorMatcher`s because that's how the previous testing was working and it still seemed to get the same result as the `Matcher`. If that's off or you feel strongly, let me know and we can go over it!

For the `UiProps` type, that was something I'd thought about and run into a few different ways. I like `UiProps` because it requires the test to have already passed and declared the added props prior to passing it into the test, so it's one less thing for the test to worry about. In my mind it seemed like the easiest solution, but again, if I'm off then let me know 😄